### DISCUSSION TITLE
[Roadmap] Ideas for 2022 Q4 Cycle 1 Prioritization

### DISCUSSION THREAD

**Discussion by iamronen** [opencollective/opencollective#5899] [Ideas]
This is the discussion to begin surfacing ideas for Q4S1.

How to use this discussion
Surfacing an idea looks very similar to ideas that have been shared in previous discussions we ask that you think about problems, (not solutions) and the impact of those problems. Consider the following questions:

Who is your user?
What are they trying to achieve?
It's useful to use the following format when trying to reflect this:

When ____ , I want to ___ , so I can ___

We ask submitters to select problems that have both a high degree of impact and a high degree of alignment with our focus areas. If there's an existing issue, please reference it, if not please [click here to create one](https://github.com/opencollective/opencollective/issues/new?assignees=&labels=&template=ideas--irritations-and-needs.md&title=).

We encourage everyone to surface ideas as early as possible so that we can start working toward a project pitch.

During the sprint (in parallel with development) we work together to create a pitch that can then be reviewed in prioritisation.

If you'd like to read more about pitches, or generally find out more about our preferred approach to development check out [Shape Up by Ryan Singer](https://basecamp.com/shapeup) (it's free)

**Comment by alanna** [comment]
## Do not allow Hosts to hide their hosted Collectives #5278

I don't think we should wait on #5457 - let's prevent disabling Contributions for active hosts now

**Comment by alanna** [comment]
## Record merchant account timestamps #5817

OCF really needs this for accounting/compliance.

**Comment by znarf** [comment]
## Ledger improvements https://github.com/opencollective/opencollective/issues/5509

It's time to look back at the ledger and clear some technical debt before the start of the new year!

**Comment by znarf** [comment]
## Expense based Contributions between Hosts https://github.com/opencollective/opencollective/issues/5100

Let's make sure money can flow easily between Hosts.

**Comment by kaylarep** [comment]
## e2c Comms Work

We could do a better job surfacing the ongoing conversations around e2c, and making sure the conversations appear as active as they genuinely are:
- https://github.com/opencollective/opencollective/issues/5318
- https://github.com/opencollective/opencollective/issues/5275
- https://github.com/opencollective/opencollective/issues/5268

**Comment by alanna** [comment]
## Add local tax support to "Add funds" (VAT, GST) #5927

Relatively small change to fix a painful issue that has no workaround and the dev team has to process manually.

**Reply by Betree** [reply]
+1 as fixing these in production is tedious and error-prone

**Reply by Betree** [reply]
I would really like us to also include these tax-related issues that cause errors in accounting:
- #4366
- #4492
- #5438

If we have the appetite for it, the following issues could also improve the general experience and reliability of taxes:
- #4493
- #3570 (would help us to stay up to date with the changes in legislation and allow us to implement new taxes more easily)

Finally, we have this **Track taxes as separate transactions #5181** issue which aligns with @znarf's [proposal](https://github.com/opencollective/opencollective/discussions/5899?sort=top#discussioncomment-3663892) above to work on ledger improvements. It is a complex one though which in my opinion is less urgent than the ones mentioned above.

Working on improving our tax support is in good alignment with developing OC Europe and having more fiscal hosts in other countries. Australia hosts are already asking for GST (#6076).

For more tax-related issues: https://github.com/opencollective/opencollective/projects/8

**Comment by kaylarep** [comment]
## https://github.com/opencollective/opencollective/issues/5815 Add Ask for Collective slug/url in 'Contact Us' form

I think this should be a simple fix that would really assist the support team

**Comment by znarf** [comment]
## https://github.com/opencollective/opencollective/issues/3781 Page in user settings to manage sessions

An important security feature.

**Comment by znarf** [comment]
## https://github.com/opencollective/opencollective/issues/5736 Revamp "Applications" and replace with "Personal Tokens"

This is getting old and ugly. And "scope" support is much needed. Would be nice improvement for Developers.

**Reply by BenJam** [reply]
where are we with the interoperability/developer integrations story? do we have any more interest?

**Comment by alanna** [comment]
## Host admin can set date when confirming or adding contribution #5693

This change would help a lot with accounting a bookkeeping practices.

**Reply by iamronen** [reply]
We (@natehn @alinamanko @poohlaga and myself) are preparing a pitch (should be ready for this prioritization) for [pending contributions](https://www.loomio.com/d/73ajeYAr/pending-contributions) and this can be bundled into that..

**Reply by alanna** [reply]
@iamronen that would be great for pending contributions - we also need it for added funds.

**Comment by BenJam** [comment]
## Pledges Redux

I'm adding this mainly to get feedback on the idea of revitalising pledges in order to solve some of the fractalization issues that exist and to encourage more projects — across all hosts — to adopt transparent financial management. 

Please dive into the discussion at https://github.com/opencollective/opencollective/discussions/6027

**Reply by BenJam** [reply]
/me crickets. 

Literally no one wants to talk about pledges?

**Reply by Betree** [reply]
We need more discussion before we can start implementing something, but I would be interested in having this discussion during the sprint.

**Reply by BenJam** [reply]
yup, @Betree I put in in here to prompt discussion ta!

**Comment by BenJam** [comment]
## Users cannot approve their own expenses as an enforceable policy #6075

A while back we asked that this feature be implemented as something a host could enact as policy, instead it was argued that we should implement it as a good will policy tool for collectives. This isn't working for us. 

Given recent events we have begun paying more attention to the small (2%) of transactions that are considerable sums. In these cases _any_ signal that we can get that others in a collective approve of expenses eases the burden on us. Without it we must try to track down historical behaviours and/or wait and _hope_ that someone else in the community notices. For example ([private link](https://opencollective.slack.com/archives/GPEHHTM4G/p1664848426928199))

_Edit: apologize, removed a [Bernie Sanders meme](https://user-images.githubusercontent.com/158833/194075821-27790c04-50c4-4c52-80fb-e21e4485aea9.png) which made its time and could unfairly attract attention when looking at all proposals --François_

**Reply by BenJam** [reply]
dammit @znarf is onto my tactics. but honestly, we have a problem with almost every transaction >$10,000 being self-approved and it's really impacting our confidence in making these payouts. It's also delaying the payment of these considerable (and therefore meaningful) payouts.

**Comment by alanna** [comment]
## Public Collective activity timeline #5767

My understanding is the current activity log work is focused on a more back-end feature mainly used by admins. I think there's big potential for a more public version of it used for community accountability and trust-building, funder due diligence, and many other functions that are only possible if it's publicly transparent. I don't imagine we'd include as much detail as the admin version of the log and we'd need to make it more user-friendly and pretty.

**Reply by Betree** [reply]
This feels to me like it should be part of a boarder collective page iteration, as proposed by @Memo-Es in https://github.com/opencollective/opencollective/discussions/5899#discussioncomment-3907002

**Comment by alanna** [comment]
## Get OCF's custom application form ready to replicate #5354

I am assuming we are still looking for a cheap MVP in this area, but there has also been a suggestion to do a custom form builder feature or integration with form builder(s) to enable this. OCEU, Gift Collective, OCNZ, and other hosts would benefit a lot from not having to juggle an external application form.

**Reply by BenJam** [reply]
OCEU used a [typeform for their application process](https://github.com/opencollective/opencollective/issues/5354#issuecomment-1269588444), which seems to work for smaller-scale (not tens or hundreds of applications a day).

**Reply by alanna** [reply]
Yeah sure using an external form is what every host besides OSC and OCF has to do, but it's very clunky and means none of those hosts can use the integrated application tracking and approval features.

**Reply by BenJam** [reply]
but we could grab the applications from the API and populate them our side with the data that was collected. 

This is an example of why we're trying to spend a little more time considering before prioritising something for development, there are alternatives.

**Comment by shannondwray** [comment]
## I'm concerned about the speed of our website - #6036

**Reply by Memo-Es** [reply]
+100

**Reply by Betree** [reply]
We already started working on that one during the sprint by introducing better monitoring tools and simple enhancements. We're still monitoring, but we're seeing major improvements already.

**Reply by BenJam** [reply]
++ this seems like it's mostly a problem with our own hosts than it is with individual collectives, but we should continue to monitor. @Betree is there a benchmark that we have for 50/95/99% of page load times, and where are the pages that are failing?

**Comment by poohlaga** [comment]
## Make bank payments on the platform easier [#4927 ](https://github.com/opencollective/opencollective/issues/4927)

Bank Transactions get incorrectly processed or ‘go missing’. Manual contributions (via bank transfer) arrive in the bank, are processed in QB but do not get represented on OC and are not allocated to the collectives.
Customers/donors are writing support tickets asking where their donations are:
https://opencollective.freshdesk.com/a/tickets/121933
https://opencollective.freshdesk.com/a/tickets/109998 

Also, manual bank transfers arrive in the bank but are meant for a different collective
https://opencollective.freshdesk.com/a/tickets/111675

I'd like us to look at the existing functionality of Stripe ACH processing to make accepting bank payments easier: https://github.com/opencollective/opencollective/issues/4927 

The benefits would be:
* Save admin time doing the matching, hunting down payments in multiple systems and correcting errors
* Less errors would improve speed and accuracy of manual bank workflow and trust from donors

**Reply by iamronen** [reply]
@poohlaga I've opened a Loomio thread and expanded on context:
https://www.loomio.com/d/cQHlRTmg/automating-bank-transfers

Also, note from internal OSC conversation: @BenJam expressed preference for SEPA over ACH because ACH is limited to USA and it may be that US banking is able to interface with SEPA.

**Reply by BenJam** [reply]
just pulling out that, at the moment around 3% of transactions (or about $300,000) for Open Source Collective are bank transfers, so for us it's worth it to improve this process.

**Comment by aerugo** [comment]
## Metadata on expenses #6038

Allow storing metadata on expenses that are intended to be read and edited by other third party services through the admin API key, and allow querying expenses through the API filtered by this metadata.

This small change to expenses requires no changes to the frontend and only some small tweaks to the backend. It would enable third-party services like Cobudget to display and submit expenses in a way that can be mapped onto other platform hierarchies.

**Comment by alanna** [comment]
## Attach files/notes to Collective #4894

As OCF's services are getting more complex and we're scaling up, it's getting too hard to rely on host admins knowing certain details as expenses are paid and other decisions need to be made, for example "we have approved xyz exception to normal protocols" "contract in place for abc" "need to follow up X on Y date". We need some way to easily connect extra notes and documents. It's also needed for audits because later - they need to be able to see the context related to a Collective clearly.

**Comment by alanna** [comment]
## Improve expanded budget section #5967

I think it's worth doing one more iteration on this before we move on. Right now, it's especially not usable for Hosts because Host fee income is not represented, generating very misleading numbers (like, we can't even turn it one for most hosts because it will not be a good representation of our budget). There have also been a few good improvement ideas and we should do the low hanging fruit to make this really cool feature more solid.

**Reply by BenJam** [reply]
I'm just here to say yes I think we should spend some time honing the budget vis but:

>especially not usable for Host

 The feature isn't really intended for hosts, it's for conveying to contributors and community members what's going on with the collectives they support. Personaly I couldn't care less about OSC's budget and professionally, I haven't enabled it

**Comment by alanna** [comment]
## Manually subscribe to Updates #1445

People continue to try to find ways to hack around this, and it's a real gap in our support for the comms and outreach side of enabling our Collectives' work. I think we should properly support it instead of living with workarounds and barriers.

**Reply by BenJam** [reply]
would #1895 be a viable alternative to this?

**Comment by iamronen** [comment]
## Improve the way we record uploaded files #3753

This feels like an inexpensive and overdue upgrade to the way we store files. 
This will have even more impact if we are going to introduce files to more contexts (eg: #4894 and perhaps pending contributions).

**Comment by alanna** [comment]
## Changelog updates or popup notifications for Hosts to notify Collectives #4708

OCF is leaning a lot more into participatory process with Collectives and needs better tools to reach them, and we're coming out with new policies and services they need to hear about. This extra feature on top of Updates would be very helpful.

**Comment by alanna** [comment]
## Internal memo field on add funds #6049

This would help with accounting and anti-fraud internal controls.

**Reply by alanna** [reply]
This is done already! Thanks @SudharakaP

**Comment by alanna** [comment]
## Allow all admins to see draft expenses #5229

This issue gets in the way of troubleshooting.

**Comment by znarf** [comment]
## Alternative Captcha provider https://github.com/opencollective/opencollective/issues/6060

We're hearing some frustration from contributors on support channels.

**Comment by Memo-Es** [comment]
## Onboarding flow - Find a host https://github.com/opencollective/opencollective/issues/5672 

Follow-up with the work to improve our onboarding flow, specifically with the experience of finding a host embedded inside the onboarding flow for Collective admins.

We said we wanted to make the onboarding flow a priority, and we have only done #5399; I propose to take on 2 or 3 more projects from https://github.com/opencollective/opencollective/issues/4829#issuecomment-1148103264 before this year is over.

**Reply by Betree** [reply]
+1. The new sign-in experience feels great, but we still have some work to do to make the onboarding a pleasant experience for collectives.

**Comment by Memo-Es** [comment]
## Contribution flow improvements

Some time ago, we prioritized this project https://github.com/opencollective/opencollective/issues/4893. It was split into https://github.com/opencollective/opencollective/issues/5631, and design research is documented here https://github.com/opencollective/opencollective/issues/5732.
So this is another attempt not to leave things in the prioritization limbo and schedule some time to get done what we said was important months ago.

**Reply by Betree** [reply]
There was a lot of feedback after I demoed https://github.com/opencollective/opencollective-frontend/pull/7992 and it stayed in limbo because I was not able to find a good compromise for a releasable MVP. I'd like to iterate on it to have something out, even if it's something simplified.

**Comment by Memo-Es** [comment]
## Profile page iteration (Continuation) #5680 

We need to follow up on this project so it does not lose momentum; we will do the first round of designs this sprint, but there is still a lot to do, and I think we need to involve the dev team starting this next sprint cycle.

**Comment by Memo-Es** [comment]
## Improve storytelling for Collectives https://github.com/opencollective/opencollective/issues/4823

We prioritized this project earlier this year, and it's a great idea, IMO. The design is ready; we just need to implement it.

**Comment by shannondwray** [comment]
## Confirm your credit card email looks phishy  #5812 

Common support request currently - very confusing for users

**Reply by Betree** [reply]
Big +1 on this one. We got many user requests and I believe there's a real (yet difficult to measure) impact on our revenue from users who just won't continue their recurring contributions because of that.

**Comment by iamronen** [comment]
## Pending Contributions #5769 

This issue was originally proposed by @natehn 
During Q3S2 a team including @natehn @poohlaga @alinamanko and myself have explored this subject in depth and have formed a proposal for a project to transform pending contributions into a valuable platform feature.

**Reply by natehn** [reply]
So excited about the possibilities of this one.

**Reply by alanna** [reply]
This is great. One challenge we're tacking right now at OCF is more closely aligning OC platform data with bank transaction data for accounting, and these improvements would help a lot.

**Reply by alinamanko** [reply]
Would love to see this happening! Would make the host admin lives SO much easier if we could make it work as intended

**Comment by shannondwray** [comment]
### Dedicated page for users to see all submitted expenses #3749 

Common ask in support especially caused drama when the paypal outage happened recently and users had trouble finding their submitted expenses 

- #5629 
- #5024

**Comment by Betree** [comment]
## Enable new PayPal integration for all hosts #4210 

Also includes https://github.com/opencollective/opencollective/issues/5923.

- We're losing a lot of time in support enabling the new PayPal integration
- It's a poor experience for users to have to contact us for enabling VS enabling directly from the interface
- The legacy adaptive is causing troubles and cost time in dev & support
- We've kept the PayPal implementations (payment & payout) in this "half baked" state for too long

**Comment by Betree** [comment]
## Redesign the page to edit tiers #3438 

The page is actually implemented, we just need to figure out a release plan:
- Address existing feedback from https://opencollective.slack.com/archives/C044GNQ7WBX/p1664521259527539
- Open a feedback issue, a docs issue
- Enable the feature for beta release on internal profiles & make an announcement to propose a release date & collect feedback

**Reply by iamronen** [reply]
@Betree can this be completed during this week and the transition week or do you feel this requires space in (and prioritization for) the next cycle?

**Reply by BenJam** [reply]
I'll do this.

**Reply by Betree** [reply]
@iamronen I think it's not too far from being completed (at least for the "internal release" phase) so it is possible that we'll be done by the end of the week. I just wanted to surface this issue as it would be sad to forget it next sprint, given that it is 90% finished; and we'll probably have some small followups once released.

**Comment by iamronen** [comment]
## Horizons: funder oriented discovery experience of what is happening in the OpenCollective ecology #6073 
This has been in preparation for the last few weeks (details in the issue) and is now being pitched for implementation.

**Comment by Memo-Es** [comment]
## https://github.com/opencollective/opencollective/issues/6077 Tags should be required when creating a Collective
This is a follow-up of https://github.com/opencollective/opencollective/issues/4764. We are investing a considerable amount of resources in creating new features such as the new discovery [(a.k.a. Collective weather report)](https://docs.google.com/document/d/1Eh_vbV7sK9FQ5HmS0P9Zqz3LReOgTdMf8BJJUcHGpI8/edit?usp=sharing), the Funders Contributors' dashboard, and the search-related targets like #3892 #3782 https://github.com/opencollective/opencollective/issues/5793 https://github.com/opencollective/opencollective/issues/5809 .
It is also necessary for fiscal hosts admin teams to have a better understandment of the communities they host and a cleaner visualization of categories.

**Comment by iamronen** [comment]
## Transform E2C page into a CMS

Currently changes to the E2C page require intervention of a designer and possibly an engineer.
This inhibits the ability to publish content.
I suggest we transform the E2C page into a minisite driven by a CMS that would allow non-technical users to make content changes.

**Reply by Memo-Es** [reply]
Just to mention here that our entire marketing system (the marketing pages) should be a headless CMS that use https://jamstack.org/ or something similar so its easier to make changes/put up new pages easily

**Reply by kaylarep** [reply]
Sounds cool to me!

**Reply by znarf** [reply]
I think we should try a bit harder to update the site with the tools and the processes we have right now.

**Reply by iamronen** [reply]
@znarf I don't think that is a sustainable approach ... I think that:
1. Engineering is too expensive to disrupt work with site content updates (unless you want to allocate this to the support engineer, though even then I doubt this is effective).
2. Prioritizing site updates is not feasible since updating contents needs to be timely and sometimes reactive.
3. Content publishing and modifications should be free to evolve without being dependent on and held back by engineering availability.

**Reply by flickz** [reply]
I agree with @iamronen although the content of other marketing pages hardly changes except when we do a general revamp. I'm also happy to explore and draft how the current headless CMS solutions fit into our current system.

**Comment by Betree** [comment]
## Introduce a centralized authentication permissions library #6085

The recent introduction of OAuth, 2FA (and fraud protection?) have made our mutations quite messy because of the numerous permission checks that need to take place. It is error prone (easy to forget something) and won't help us to enforce best practices in the team.

I'm expecting this project to have a limited complexity, the most expensive part is probably to discuss and agree on specifications.

**Comment by BenJam** [comment]
quick update on prioritisation following a conversation between myself and @znarf:

**external commitments**
We have committed to delivering #6073 to an external funder (and Open Collective Foundation is covering the cost of 1 developer for a whole year for this) and we need to show some progress before the end of the year. 

**strategic priorities**
We're currently reviewing our strategy and will be publishing something for comments later this or early next week. But the pattern that is emerging is that we believe we should focus on our foundation, and on growing existing first-party hosts. We did not use this to select any projects. 

**Tactical priorities**
There were 13 surveys completed, the results of which selected three issues that were consistently in the top 1/3rd of respondents' rankings: #5812, #3749, and #4927. Every other issue was ranked between 10 and 20 meaning that most people placed it mid-board or equally many people placed it in the top 1/3rd as did the bottom. 

**Engineering, Support and Ops**
With these projects prioritised we asked what engineering needed to tackle as a priority, we selected #5509. Finally we looked at projects that would have an impact on support and operations, we selected #5815. 

**Extra capacity**
If we have any extra time this sprint (or we feel like spending some time over the holidays 19th Dec - 3nd Jan) we thought that #5229 would support Open Collective Foundation and #6075 to support Open Source Collective would be a good division. Finally, we thought revisiting the feedback on #5967 and implementing the top 1-2 items would support both Open Source Collective and Open Collective Foundation.
